### DISCUSSION TITLE
Update to RavenDB cloud caused my index to break?

### DISCUSSION THREAD

**Discussion by nickchampion** [ravendb/ravendb#15905] [General]
Hi, I just started getting an error on one of our indexes today with no changes from our side. The error is essentially saying we cant query on a field thats not indexed, the odd thing is the fields appear to be indexed. 

See the image below showing the raw indexed values

<img width="1994" alt="image" src="https://user-images.githubusercontent.com/3949005/221586562-1a6ea597-9cf0-44a2-bc11-214a99822e3a.png">

When I run this query 

`from index 'Storages' where organisationId = 'organisations/1-A'`

I get this error

<img width="326" alt="image" src="https://user-images.githubusercontent.com/3949005/221586754-57c4424c-e745-43f4-91e5-322a90cca827.png">

I have fixed this by changing the way we return fields in the index but would like to understand what happened.

The index in question is a Javascript index which generates custom fields and uses Object.assign({}, {}) to return the fields. Something like this where the fields variable are the dynamic fields

```
return Object.assign(fields, {
  name: storage.name,
  type: storage.type,
  storageId: id(storage),
})
```

I changed this to 

```
const result = Object.assign(fields, {
  name: storage.name,
  type: storage.type,
  storageId: id(storage),
})

return result
```

Which resolved it.

Any idea whats going on here? Was there a new release today that could have caused this issue?

I can email a certificate to support with access to this database and the index if someone wants to take a look at the behaviour.#

Thanks

**Comment by arekpalinski** [comment]
Hi Nick,

The issue got fixed already and we'll install the fix in RavenDB Cloud today: https://github.com/ravendb/ravendb/pull/15822

**Reply by nickchampion** [reply]
Ok thanks @arekpalinski
